File: q2.py
#2 - Salve em um arquivo .csv o first_name e o gender de cada pessoa. E imprima na tela a média das idades;

import csv

#Abertura do arquivo de dados na var dados
with open('src/data.csv') as dados:
    
    #Configurando o signo de tabulacao
    leitor = csv.DictReader(dados, delimiter=',')
    
    #Contados de linhas
    cont = 0
    #Acumulador de idades 
    idades = 0
    
    #Cria o arquivo de saida
    with open('results/results_q2.csv', 'w') as saida:
  
        #Monta as colunas da base de dados
        colunas = ['first_name', 'gender']
    
        #Vincula o arquivo a um buffer de edicao, configura as conlunas e o delimitador
        maquina = csv.DictWriter(saida, fieldnames = colunas, delimiter = ',', lineterminator = '\n')
        maquina.writeheader()
    
        #Iterator para percorrer o arquivo de dados aberto
        for linha in leitor:
            cont += 1
            idades += int(linha['age'])
            
            #Escreve a no arquivo de saida os dados lidos da base de dados
            maquina.writerow({'first_name': linha['first_name'], 'gender': linha['gender']})

print ("media de idades: " + str(idades/cont))

File: q2.java
import java.util.*;

//2 - Salve em um arquivo .csv o first_name e o gender de cada pessoa. E imprima na tela a média das idades;

public class Teste{
	public static void main(String []args){
		//Endereco do arquivo	
		String end = ();
		
		double media = 0;
		int cont = 0;
		String[] dados = new String();
		
		try{
			//Objeto de escrita
			FilerWriter base = new FilerWriter("dados.csv")
			PrintWriter maquina = new PrintWriter(base);
			
			//Buffer de leitura	abertura		
			BufferedReader arquivo = new BufferedReader(new FileReader ("data (1).csv");
			
			//Leitura do arquivo
			while(arquivo.ready()){
				//Linha atual, cursor
				String linha = arquivo.readLine();
				
				//Quebra de dados
				dados = linhas.split(",");
				
				//Contagem de individuos
				++cont;
				
				//Acumulo das idades
				media += dados[3];
				
				//Salvar dados no arquivo de saida
				maquina.print(dados[0]+ "," + dados[2] + "\n");
			}
			
			//Encerrar buffers
			arquivo.close();
			maquina.close();
			
			//Saida da media
			media = media/cont;
			System.out.printl("A media das idades e: ");
		}catch(IOException ioe){
			ioe.printStackTrace();
		} 
	}
}